s3store/s3store.go

```diff
@@ -536,3 +549,21 @@ func isAwsError(err error, code string) bool {
 	}
 	return false
 }
+
+func (store S3Store) CalcPartSize(size *int64) (effectivePartSize int64) {
+	switch {
+	case *size > store.MaxPartSize*10000:
+		effectivePartSize = store.MaxPartSize
+	case *size <= store.MinPartSize:
+		effectivePartSize = store.MinPartSize
+	case *size/10000 < store.MinPartSize:
+		effectivePartSize = store.MinPartSize
+	case int64(math.Mod(float64(*size), 10000)) == 0:
+		effectivePartSize = *size / 10000
+	case *size/9999 >= store.MaxPartSize:
+		effectivePartSize = store.MaxPartSize
+	default:
+		effectivePartSize = *size / 9999
+	}
+	return effectivePartSize
```

Comment:
I am sorry but I am lost with this switch case here. Could you explain why you ditched your proposed algorithm from https://github.com/tus/tusd/issues/149?